woocommerce/woocommerce
# [Accessibility] Make the matching variations alert a live region

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





Closes #43613 .



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Set up the store by importing the sample data.
2. On the admin, edit the Hoodie product.
3. Go to the Attributes tab and add the following options. Some of these options are not created by default, so you need to do it using the "Create value" button.
    - **Color**: Blue, Green, Red, and Yellow.
    - **Size**: 5XL, 4XL, 3XL, 2XL, XL, L, M, and S.
    
![image](https://github.com/user-attachments/assets/e2b1b680-fae0-4c19-a6a1-be450815d9f3)

4. Save the attributes.
5. Go to the Variation tab.
6. If there is any variation already created, delete them.
7. Click on the "Generate variations" button.
8. Delete one of the variations. For example, the Blue color with the 5XL size.
9. Click on the variation rows and confirm each one has a price. If some of them don't have a price, add it.
10. Save changes and update the product.
11. Go to the product page on the frontend.
12. Turn on the screen reader.
13. Select the Blue color with the 5XL size.
14. Verify the screen reader announces "Sorry, no products matched your selection. Please choose a different combination."



### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
